### DISCUSSION TITLE
Value marshalling in TypeSpec 0.57^

### DISCUSSION THREAD

**Discussion by Onokaev** [microsoft/typespec#3677] [General]
I am working on upgrading our TypeSpec version to the latest version and fixing all breaking changes. However, I have a decorator with the alias '$flags'. This is causing conflicts when I apply the suggestion here:

```typescript
Deprecated: Parameter name of decorator @function is using legacy marshalling but is accepting null as a type.
This will change in the future.
Add `export const $flags = {decoratorArgMarshalling: "new"}}` to your library to opt-in to the new marshalling behavior.
> 206 | extern dec function(target: Operation, name?: valueof string | null, isComposable?: valueof boolean);
```

Does the exported constant have to be $flags?

**Comment by Onokaev** [comment]
Here is a snapshot of this decorator and the error I am getting when I upgrade to 0.57
![image](https://github.com/microsoft/typespec/assets/45680252/01329ca5-48d2-41d9-930c-4b45549ae592)

**Comment by Onokaev** [comment]
Just a gentle ping @timotheeguerin

**Comment by timotheeguerin** [comment]
Seems to be a limitation we have today filed an issue https://github.com/microsoft/typespec/issues/3719
